Training24/Test.c

```diff
@@ -1,14 +1,163 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Test.c - The program sorts integers with insertion sort and searches for numbers using binary search,
+// with options for test cases and user input.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>
+#include <stdlib.h>
+#include <conio.h>
+#include <stdbool.h>
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+static void DisplayArray (IndexedElement arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i].value);
+   printf ("|\n");
+}
+
+/// <summary>Function to check if the array is sorted.</summary>
+static bool IsSorted (IndexedElement arr[], int size) {
+   for (int i = 1; i < size; i++) if (arr[i - 1].value > arr[i].value) return false; // Not sorted
+   return true; // Sorted
+}
+
+/// <summary>Checks if the input string is a valid integer (digits only).</summary>
+static int IsValidInteger (const char* str) {
+   if (str == NULL || *str == '\n' || *str == '\0') return 0;
+   while (isspace (*str)) str++;
+   if (*str == '+' || *str == '-') str++;
+   for (; *str != '\0' && *str != '\n'; str++) if (!isdigit (*str)) return 0;
+   return 1;
+}
+
+/// <summary>Function to format and print test results.</summary>
+static void PrintTestCase (int numTests, IndexedElement inputs[], int size, int searchElement) {
+   IndexedElement sortedArray[50];

```

Comment:
No need to define a new struct, just simply use the array instead and Display of original index is unnecessary as you're displaying the whole input array itself.